### DISCUSSION TITLE
Support adding filters in lsp

### DISCUSSION THREAD

**Discussion by wrvsrx** [zk-org/zk#329] [Ideas]
### Verify

- [X] I searched the existing discussions for similar ideas

### Share your idea or feature request

Sometimes, I might archive some notes by giving them "archived" tag and I can use a shell alias such as `zk list -t 'NOT archived'` to filter them out. When I edit notes, I also want lsp filter them out when completing. Can zk support add filters in lsp?

**Comment by mickael-menu** [comment]
I'm not sure I understand exactly what you want.

Are you using Neovim? You can call something like that: `:ZkNotes {tags = {'NOT archived'}}`.

**Comment by wrvsrx** [comment]
For example, I have note A, B, C. When editing note C with neovim or vscode, if I input `[[`, lsp will give me some completion items.

Sometimes, I want only A appear in my completion items. Therefore I hope I can config lsp to filter some unwanted notes.

**Reply by mickael-menu** [reply]
I see, I'm not sure this is possible with the LSP completion items, unless we declare dynamically a different trigger (`[[`) per set of notes. I don't think I want to go down that road with `zk`.

However, ideally an equivalent to Neovim's `:ZkNotes` should be added to VSCode, using the native UI, to pick and insert a note.

**Reply by wrvsrx** [reply]
I just want an option in the LSP section that can help me filter out certain notes that I don't want to appear, such as notes that have been tagged as "useless" or "archived". If you're not very interested in this, may I ask if you would accept a PR from me later?

**Reply by mickael-menu** [reply]
Ha I see, so it would be a set of filtering option that would be always on for all LSP completion items? Yes that would be fine I guess.

A PR would be welcome, I probably won't tackle this myself.

I suggest adding a new `note-filter` property in the [`lsp.completion` config section](https://github.com/mickael-menu/zk/blob/main/docs/config-lsp.md#complete-example). The syntax could be the same as the one from the [`filter` config section](https://github.com/mickael-menu/zk/blob/main/docs/config-filter.md#named-filter), so basically the CLI flags. You would need to use Kong to parse the filtering flags, see `internal/cli/filtering.go`.
